rust/src/index.rs

```diff
@@ -43,6 +43,8 @@ use self::vector::{build_vector_index, VectorIndexParams};
 pub(crate) trait Index: Send + Sync {
     /// Cast to [Any].
     fn as_any(&self) -> &dyn Any;
+
+    fn statistics(&self) -> Result<String>;
```

Comment:
I think we probably need to return `serde_json::Value` here instead, so that we don't created JSON within JSON.

```suggestion
    // TODO: if we ever make this public, do so in such a way that `serde_json`
    // isn't exposed at the interface. That way mismatched versions isn't an issue.
    fn statistics(&self) -> Result<serde_json::Value>;
```